### PR TITLE
Listing of datasets held in socrata

### PR DESCRIPTION
**Pull request by pschmied**
Hello and thank you for writing this package. I was just about to start writing something similar when I found yours.

One use case I have is to iterate over datasets. In order to facilitate that, I wrote a function to return a dataframe containing titles and descriptions of datasets. 

Cheers,
Peter

### CODE REVIEW

**Comment by tomschenkjr** [comment]
@pschmied - thank you for your work on this. We like this concept and functionality, will be a great addition to the library.

We've been doing some initial testing and noticed there seems to be errors in the data being returned from the OData call. This isn't an issue with the library, but seems to be related to what is being returned in the XML. Namely, fewer results are being returned with OData. When I checked Chicago's feed, the number of datasets was a couple hundred less than expected.

Conversely, every Socrata portal has a data.json feed at ```http://portal.example.com/data.json```. This seems to return the correct number of results. I am reaching out to the Socrata support team (/cc @chrismetcalf @alaurenz) to see if this can be explained. 

So, we can see if there is some error returning fewer values for OData. Alternatively, it may be advisable to reference ```data.json``` file. Again, nothing wrong with the library, but want to make sure OData is the best service to reference.

**Comment by pschmied** [comment]
Thank you for the reply!

First off, I'm definitely not wed to using the OData protocol. That's
just what I was able to find in my googling. Had I known about
data.json, that would have been my first port of call.

As for the number of results, I found something similar with City of
Seattle's portal. I was able to track the difference down to some of the
data being in a "federated domain." However, if the json api is smart
enough to return all datasets across federated domains, then that's
likely much better.

Happy to update things to use json here in a little bit.

Cheers,
Peter


Tom Schenk Jr writes:

**Comment by tomschenkjr** [comment]
Hi Peter-

The issues I saw with the City of Chicago site, I think, is a bit more than federated datasets.

Also, there is a data quality issue on the OData feed. A number of the URLs (the so-called four-by-four) are incorrect on OData, but correct in data.json.

I'm interested in the answer nevertheless, but would be good if this was switched to data.json (I think that would remove one package dependency).

Thanks for your patience!

/cc @Chicago/open-gov-team as FYI to the difference between the two feeds.

**Comment by pschmied** [comment]
No worries. I've gone ahead and modified the code to work with json (much
less work), and will send a new PR once I'm no longer on a bus.

**Comment by pschmied** [comment]
Alright, it looks like this PR has been updated with the changes to json. Much better / easier.

**Comment by chrismetcalf** [comment]
@tomschenkjr For the count of datasets via OData, are you looking at https://data.cityofchicago.org/OData.svc?

`/OData.svc` returns 328, which matches with the count in the catalog when you [filter for just "datasets"](https://data.cityofchicago.org/browse?limitTo=datasets&utf8=%E2%9C%93)

`/data.json` returns 533 (counting one self-referential entry back to `/data.json`), and also includes resources that aren't accessible via the OData or SODA APIs, like downloadable KML files and the like. That may affect whether or not they'll be useful for RSocrata too.

**Comment by tomschenkjr** [comment]
@chrismetcalf - Ah, that makes perfect sense. It should have dawned on me that I was only looking at tabular data. Thanks, Chris.

File: DESCRIPTION [hunk 1]
```diff
@@ -6,14 +6,14 @@ Description: Provided with a Socrata dataset resource URL,
     returns an R data frame.
     Converts dates to POSIX format.
     Manages throttling by Socrata.
-Version: 1.5.1
-Date: 2014-12-08
+Version: 1.5.2
```
Comment (tomschenkjr) [inline DESCRIPTION]: We're going accept this as a beta for version 1.6.0, which will be submitted to CRAN in a month or so. Can you adjust this to "1.6.0-beta"?

Comment (pschmied) [inline DESCRIPTION]: Just pushed an updated version with the adjusted version number.

Tom Schenk Jr writes:

**Pull request merged**